### DISCUSSION TITLE
全角に替わる基準単位の議論

### DISCUSSION THREAD

**Discussion by kidayasuo** [w3c/jlreq-d#59] [General]
下の文書に対する議論

https://github.com/w3c/jlreq-d/blob/gh-pages/drafts/.%E5%85%A8%E8%A7%92%E3%81%AB%E6%9B%BF%E3%82%8F%E3%82%8B%E5%8D%98%E4%BD%8D.md

@w3c/jlreq

**Comment by kidayasuo** [comment]
敏 @KobayashiToshi 先生から[このバージョンの文書](https://github.com/w3c/jlreq-d/blob/f0dcb4114c877c685f60a92135f1a4505ce40869/drafts/.全角に替わる単位.md)に直接いただいたコメントをここに移動しました：

Jlreq-dでは【binn：、字詰め方向（文字を行に配置していく方向）において，幅を示す相対的な単位として】従来から使われてきた「全角（=em square?）【binn：全角（em又はfull-width）】」に替えて、「全幅（fullwidth）」という単位を次のように定義し、使用する。【binn：従来から使われてきた「全角」が何を差すか不明なのですが，JLReqでといえば，full-widthとなり，一般的にいえば“em”ともいえる．】

全幅【binn：とはその場所の→：そこに使用されている】フォントの和字の字送り幅【binn：字送り幅→文字の外枠の字詰め方向のサイズ．横組では文字の外枠の左右のサイズ、縦組では文字の外枠の左右のサイズとなる】。【binn：全角空白（U+3000）グリフの文字送り方向の幅として測定する。→和字の字送り幅としては，そのフォントの和字間隔（U+3000）を使用する。JIS X 0213では“和字間隔”といっているが，わかりにくかもしれない．】

binn：字送りの定義が必要になるので，“字送り”の使用は，できれば避けたい．また，全角空白（U+3000）グリフというのも分かりにくい．また，全幅は，字送りの大きさだけを示すものではない．字幅やアキの大きさも示す．

binn：現在は，文字の外枠（つまりボディ）という説明で字の文字の配置を説明してきた．それを字送りという概念で説明することは可能であるが，文字の外枠（つまりボディ）の原点位置を決めないと，その説明はできない．また原点の位置によって説明が変わるケースがある．実際のフォントの原点の位置は，横組で変わる可能性があり，さらにフォントにより異なる可能性があるというのがNatさんの説明．現実の処理は字送りですので，字送りで説明できれば本当はよいのだが，こうした事情は，文字の外枠（つまりボディ）という概念を使えば，原点がどこであってもよく，その問題を回避できる．それよりもイメージもしやすいという利点があると思います．そもそも四分アキとか八分アキという概念を用いるということは，実際の処理ではないと思います．それは字送りで処理しているでしょう．すべてできるのに，それをしないで，分かりやすさからではないでしょうか？　また，変形した場合など，字幅という説明をしないで，字送りで，すべて説明しようと思えばできる．変形した場合など，どうしても字幅という説明を加えた方が分かりやすくなる，ということではないでしょうか．

「全幅」は従来の組版規則で使われてきた「全角」という単位を、【binn：ボディー→文字の外枠】が正方形でないフォントに対しても適用できるよう、【binn：字送り→字詰め】方向の幅に限定し一般化したものである。「全角」は正方形であって【これもやや不正確な表現】長体や扁平体の字送り幅を表現できないのに対し、全幅は【binn：字送り→字詰め】方向の幅のみを表し【binn：これは全角も同じ】、これらのフォントにも適用することができる。また、その測定方法から、漢字と仮名の幅が異なるフォントや、仮名がプロポーショナルなフォントに対しても定義することができる。【binn：適用はでき，定義はできるが，プロポーショナルなフォントでは，使い道があるかどうかはわからない，と思いますが．つまりプロポーショナルな文字や漢字と仮名の幅が異なる場合に，四分アキといったときに，全角空白（U+3000）の1/4が適切かは不明だと思います．そもそも全角空白（U+3000）の幅がどうなっているか，こうした事情ではわからないのではないか．変形や等幅のフォントでは，はっきりしているので，使い道はある．】

大多数の日本語フォントでは全幅は全角の大きさに一致する【binn：この文，やや正確さに欠けるようにも思いますが】。Unicode / CSS などでfullwidthという用語が全角文字の幅として使用されているので、全幅の英訳としてfullwidthを使用する。【この2文は別段落がよい】【binn：fullwidthという用語はどのように定義されていますか？　fullwidth charactersのことか．つまり，文字種を示すのか，それとも別のことか？　また，JLReqでは“全角”の英訳が“full-width”となっているので，混乱しないか？】

**Reply by kidayasuo** [reply]
それぞれについて、コメントします。コメントに番号をつけました：

>JLReq-dでは【1: binn：、字詰め方向（文字を行に配置していく方向）において，幅を示す相対的な単位として】従来から使われてきた「全角（=em square?）【2: binn：全角（em又はfull-width）】」に替えて、「全幅（fullwidth）」という単位を次のように定義し、使用する。【3: binn：従来から使われてきた「全角」が何を差すか不明なのですが，JLReqでといえば，full-widthとなり，一般的にいえば“em”ともいえる．】

1.  確かに全角は、正方形であることや、字詰め方向の幅、それと直角の幅、と色々に用いられます。ここでは字詰め方向の幅に限定しているので、そう書き添えようと思います。
2.  jlreq-dではこの訳語を使う、という意味でしたが、ここでカッコの中に示すのは紛らわしいですね。削除します。
3. 従来から使われてきた全角は、JLReqに限らず、正方形であることや、字詰め方向の幅、それと直角の幅、と色々に用いられます。従来fullwidth/full-widthが訳語のように使われてきましたが、widthは明確に幅ですので、この訳語はずれがある。何がずれなく等しいかというとem / em squareかと思います。

>全幅【4: binn：とはその場所の→：そこに使用されている】フォントの和字の字送り幅【5: binn：字送り幅→文字の外枠の字詰め方向のサイズ．横組では文字の外枠の左右のサイズ、縦組では文字の外枠の左右のサイズとなる】。【6: binn：全角空白（U+3000）グリフの文字送り方向の幅として測定する。→和字の字送り幅としては，そのフォントの和字間隔（U+3000）を使用する。JIS X 0213では“和字間隔”といっているが，わかりにくかもしれない．】

4. 「そこに使用されている」がより明確なので、変更します。CSSなどではその場所の属性、プロパティの一つなので、その場所の、が適当なのですが、技術的ではなく、より一般的な言い方にします。
5. 確かに字送り幅では、字間に加えたスペースも含んでしまいますね。「文字の外枠の字詰め方向のサイズ」はちょっと長いので、短くしたい気持ちがあります。文字であることは明確なので「ボディの字詰め方向の幅」ではどうでしょう。外枠、よりもボディの方が今やわかりやすいかと思いますが、どうでしょう？
6. 正式な名称「和字間隔（U+3000）」に変更します。

>7:  binn：字送りの定義が必要になるので，“字送り”の使用は，できれば避けたい．また，全角空白（U+3000）グリフというのも分かりにくい．また，全幅は，字送りの大きさだけを示すものではない．字幅やアキの大きさも示す．

7. 「字詰め方向の幅」に書き換えます。空白もグリフを持ちますので、その点は不明確にはならないかと思います。全幅は、アキの大きさなどにも使いますが、ここで述べているのは単に定義方法ですので、問題ないかと思いますがいかがでしょう。


>8: binn：現在は，文字の外枠（つまりボディ）という説明で字の文字の配置を説明してきた．それを字送りという概念で説明することは可能であるが，文字の外枠（つまりボディ）の原点位置を決めないと，その説明はできない．また原点の位置によって説明が変わるケースがある．実際のフォントの原点の位置は，横組で変わる可能性があり，さらにフォントにより異なる可能性があるというのがNatさんの説明．現実の処理は字送りですので，字送りで説明できれば本当はよいのだが，こうした事情は，文字の外枠（つまりボディ）という概念を使えば，原点がどこであってもよく，その問題を回避できる．それよりもイメージもしやすいという利点があると思います．そもそも四分アキとか八分アキという概念を用いるということは，実際の処理ではないと思います．それは字送りで処理しているでしょう．すべてできるのに，それをしないで，分かりやすさからではないでしょうか？　また，変形した場合など，字幅という説明をしないで，字送りで，すべて説明しようと思えばできる．変形した場合など，どうしても字幅という説明を加えた方が分かりやすくなる，ということではないでしょうか．

8: ご指摘ありがとうございます。よりわかりやすく、ボディの幅として表現しようと思います。この表現は物理的な活字でイメージしやすいのでその点でもより良いかと思います。原点の問題は字送りの大きさや幅とは全く独立の問題かと思います。どこが原点であろうと、文字に固有の値ですので。字送りというと字間にアキを取った場合にそれも含んでしまうので、不適当かと思いました。

>「全幅」は従来の組版規則で使われてきた「全角」という単位を、【9: binn：ボディー→文字の外枠】が正方形でないフォントに対しても適用できるよう、【10: binn：字送り→字詰め】方向の幅に限定し一般化したものである。「全角」は正方形であって【11: これもやや不正確な表現】長体や扁平体の字送り幅を表現できないのに対し、全幅は【12: binn：字送り→字詰め】方向の幅のみを表し【13: binn：これは全角も同じ】、これらのフォントにも適用することができる。また、その測定方法から、漢字と仮名の幅が異なるフォントや、仮名がプロポーショナルなフォントに対しても定義することができる。【14: binn：適用はでき，定義はできるが，プロポーショナルなフォントでは，使い道があるかどうかはわからない，と思いますが．つまりプロポーショナルな文字や漢字と仮名の幅が異なる場合に，四分アキといったときに，全角空白（U+3000）の1/4が適切かは不明だと思います．そもそも全角空白（U+3000）の幅がどうなっているか，こうした事情ではわからないのではないか．変形や等幅のフォントでは，はっきりしているので，使い道はある．】

9. ボディではだめ？　こちらの方がわかりやすいかと思いますが。どちらでも英語にするとbodyですが、日本語訳の用語としてどうでしょうね。
10. はい、字詰めにします。
11. この場所に必要十分に正確なら良いかと思いますが、誤解の恐れがある場合は訂正します。
12. 字詰めにします。
13. 全角は幅のみではなく、字詰め方向と直角な方向のサイズも表します。例えば行間に使われているのはその証左です。
14. 例えば、MS Pなんちゃら、は仮名プロポーショナルですが、それを使った時の段落行頭一字下げは、やはりその和字間隔が意図されていると考えるのです。とすると、全幅の定義は、仮名がプロポーショナルなフォントに対して、少なくとも段落行頭一字下げは適用できることになります。その他の場合に適用できるかは不明。とはいえ、プロポーショナルでも「定義」できると言っているだけで、「適用」できる、と言っているわけではないので、その誤解が生まれないように表現を考えたいと思います。

> 大多数の日本語フォントでは全幅は全角の大きさに一致する【15: binn：この文，やや正確さに欠けるようにも思いますが】。Unicode / CSS などでfullwidthという用語が全角文字の幅として使用されているので、全幅の英訳としてfullwidthを使用する。【16: この2文は別段落がよい】【17: binn：fullwidthという用語はどのように定義されていますか？　fullwidth charactersのことか．つまり，文字種を示すのか，それとも別のことか？　また，JLReqでは“全角”の英訳が“full-width”となっているので，混乱しないか？】

15. どのように不正確でしょう？
16. そうします。
17. 正確な定義がないんですよ。UnicodeでもCSSでもふわふわした言い方です。一番はっきり書いているのはCSS Text Level 4で、そこには漢字が等幅であるCJKフォントの漢字のadvance widthと書いてあります。英訳の問題は、今でも意味がずれているので、それほど混乱しないかと思っていますが、どうでしょうね。

**Reply by kidayasuo** [reply]
@KobayashiToshi アップデートしました。[アップデートしたバージョン](https://github.com/w3c/jlreq-d/blob/85f5b914b2bb09077049d4472d6d088377f334f4/drafts/.全角に替わる単位.md)。[差分](https://github.com/w3c/jlreq-d/compare/f0dcb41..85f5b91?diff=split&w=)。

- 全体にわたって「字詰め方向の幅」という表現を使うようにしました。
- 「外枠」は「ボディ」という、おそらくより知られており、そのまま英語になる用語を使っています。
- プロポーショナルな場合への言及は削除し、背景のセクションに移しました。
- 15の「大多数の日本語フォントでは全幅は全角の大きさに一致する」に対する「この文，やや正確さに欠けるようにも思いますが」はどのように不正確かわからなかったので、そのままです。

**Comment by KobayashiToshi** [comment]
>  3.  それと直角の幅、と色々に用いられます。

その意味はあります．

> 3.  em squareかと思います。

ちょっと聞きなれない言葉のように思います．

> 5． 「ボディの字詰め方向の幅」ではどうでしょう。外枠、よりもボディの方が今やわかりやすいかと思いますが、どうでしょう？

用語の問題ですが，できるだけ新たな用語を使用しないで，JLReqで使用し，それでよければ使用した方がいいように思います．新たな定義をしないですみます．“文字の外枠”“字詰め方向”“行送り方向”“字幅”は，JLReqで定義し，使用している用語です．

> 7．「字詰め方向の幅」に書き換えます。空白もグリフを持ちますので、その点は不明確にはならないかと思います。全幅は、アキの大きさなどにも使いますが、ここで述べているのは単に定義方法ですので、問題ないかと思いますがいかがでしょう。

文字の外枠の字詰め方向の幅は“字幅”です．最初に，ボディの字詰め方向の幅である字幅，又は文字の外枠の字詰め方向の幅である字幅と最初に言い，後は“字幅”とする，という言い方もあります．

空白もグリフをもっているんだ．ただ，ちょっと気になるのは，漢字や仮名のグリフは，一般に文字の外枠より小さくなるので，どうなんだろうという疑問がでるかもしれない．

> 8:ボディの幅として表現しようと思います。この表現は物理的な活字でイメージしやすいのでその点でもより良いかと思います。

ただ，実際の処理は字送りで処理していると思います．ですので，どこかで，簡単でよいので，ちょっとだけでも“文字外枠+アキモデル”と“字送りモデル”のことを触れた方がよいかもしれない．

> 9．ボディではだめ？　こちらの方がわかりやすいかと思いますが。どちらでも英語にするとbodyですが、日本語訳の用語としてどうでしょうね。

私は“文字の外枠”が最適とは思っていませんが，私がJIS X 4051に参加した時にはすでに使用されていたので，それを使っているのですが，……．もし　“ボディ”にするなら，それを最初に定義または説明しないといけない．なお，“文字の外枠”の英訳は“character frame”です．

> 11．この場所に必要十分に正確なら良いかと思いますが、誤解の恐れがある場合は訂正します。

これどこの箇所の問題ですか？

> 13．全角は幅のみではなく、字詰め方向と直角な方向のサイズも表します。例えば行間に使われているのはその証左です。

“字詰め方向と直角な方向のサイズ”に，例えば“全角行間”なんていいかたをしますので，その通りです．

> 大多数の日本語フォントでは全幅は全角の大きさに一致する

>15．どのように不正確でしょう？

全角は，JLreqでは“a 文字サイズに等しい長さの相対単位”という意味と，“b 字幅がaである文字の外枠．この場合，文字外枠は正方形になる．”という意味があるので，どちらの意味かが不明ということです．“大多数の日本語フォントでは，文字の外枠（ボディ）は正方形であるので，全幅は，その正方形の横幅でサイズであり，文字サイズとなる．”みたいな言い方もできる．

> 17．“fullwidth”

JLReqで全角の英訳に“full-width”とあるので混乱しないか，また，“fullwidth characters”の意味に理解される可能性はないといえるか（誤解される可能性がないかどうか，私にはわからないが），ということです．

**Reply by kidayasuo** [reply]
### ボディ vs 外枠
私はボディの方がわかりやすいと考えるので、この言葉を使った方が良いかと思いますが、JISから変えない方が良いという考え方も理解できます。次のミーティングで議論しましょう。

ところで、character frameという用語は私は聞いたことがありません。「外枠」をそのまま英語にしようとして生み出した和製の言葉という可能性はありますか？　山本さんに要確認ですが、一般的に使われている用語はbodyかと。

### グリフ
はい、グリフは、文字の黒いところだけではなくて、設計された空間も含んでいます。なので、スペース文字も立派なグリフですし、ラテン文字なら設計された空間も含みますし、漢字や仮名はその外枠まで、ボディ全体でグリフです。

いまふと気がつきましたが、外枠とボディは同じようで、枠線だけか、中身が詰まっているか、という結構大きな違いがありますね。大きさの議論には関係ありませんけれど。

### 字詰め方向の幅である字幅
確かにそのように定義してしまえば、毎回「字詰め方向の幅」と口はばったい言い方をせずに済みますね。そうしましょう。

### 外枠＋アキ vs 字送り、のモデル
この違いは、純粋にに用語の使い方、もしくは実装の方法かと思うのですが、どうでしょう？　そうならば特に触れる必要はないかと思います。何らか組版の考え方に影響があるのなら、触れましょう。

### 11. の不正確な表現？
> これどこの箇所の問題ですか？

ここです：「全角」は正方形であって【11: これもやや不正確な表現】

全角の「角」は、角柱、角砂糖、に見られるように四角形という意味かと思いますが、いかがでしょう？

### 全角の大きさ、という表現
「全角の**大きさ**」と「大きさ」をつけているので、「正方形の大きさ」と同じで、誤解の余地はないかと思いますが、どのように違って取られる可能性がありますか？

JLReqでは字詰め方向と直角方向の大きさとしての全角という用語を避けているとのこと。しかしX4051や他の組版の記述には使われているかと思います。特に行間や基本版面の記述に。例え、全角という用語から字詰め方向と直角方向の大きさという意味を取り除いたとしても、全角=正方形、もしくは全角=文字サイズ、という概念が残っている限り、全角を長体や扁平体の字詰め方向の幅として考えることはできません。それが前回のミーティングの結論だったかと思います。


### fullwidth または full-width という訳
「全角」の訳としてこれらの用語が使われていますが、その意味は元々の全角から大きくずれていて、文字が正方形かどうかに関わらず、等幅なCJK文字の字詰め方向の字幅を表しています。これはすなわち今回定義する「全幅」であって、逆にfullwidthの正確な和訳として「全幅」を作ったと言っても良いくらいです。なので、全幅をそのままfullwidthと訳しても混乱はなく、より正確になるかと思います。

fullwidthはまたfullwidth characterという言い方で字種を表すのに使われますが、これに関しては「全角約物」などと依然として「全角」を使わざるを得ないでしょう。

混乱の可能性は確かにあると思いますが、可能性であって、具体的に混乱が起こってから、どんな混乱が起こったかを見て対処を考えれば良いかなと考えますが、いかがでしょう？

JIS X 4051において、下のような記述がありましたので、全角の英訳としてはemで良く、かつそれが正確かと思います。
> 67) 全角（ぜんかく） 漢字1文字分の外枠。欧文の組版ではエム（em）ともいう。

**Comment by KobayashiToshi** [comment]
> character frameという用語は私は聞いたことがありません。「外枠」をそのまま英語にしようとして生み出した和製の言葉という可能性はありますか？

たぶん，そうでしょう．“文字の外枠”という用語は，そもそも組版の用語として定着したものではない．これはJIS X 4051の用語です．それをJLReqでも踏襲しただけともいえる．以下，独断が入りますが，背景を理解してもらうために，多少は不正確でしょうが，説明いたします．まず，JISの“文字の外枠”という用語は私の参加以前に決めたことですので，その経過は，よくわからない．推測ですが，従来からは“仮想ボディ”という言い方をしており，Natさんや山本さんは，“仮想ボディ”という用語を，現在の議論でも，よく使用している．しかし，“仮想ボディ”という用語は，あくまで文字を設計する際の枠であり，組版の際に，かつての活字の台，つまりボディに該当する用語としてはふさわしくない，という考え方があって，“仮想ボディ”の代わりに考えられたように，私は理解している．私は，もっぱらこの用語を使用しているので，なじんではいますが，……，さて，ということです．

> 「全幅」は従来の組版規則で使われてきた「全角」という単位を、ボディが正方形でないフォントに対しても適用できるよう、字詰め方向の幅に限定し一般化したものである。「全角」は正方形であって、正方形のボディを持たない長体や扁平体の……

長さの単位の話なのに，面積が問題である“正方形”という用語は出てくるのに，少し違和感を感じたたということです．以下のようにしてもよい．

「全角」という単位は，正方形のボディ（文字の外枠）を前提としているので、ボディが正方形でないフォントに……

> 外枠＋アキ vs 字送り、のモデル

木田さんが気にしないのであれば，いれなくてよい．

> 大多数の日本語フォントでは全幅は全角の大きさに一致する。

では，以下のようではどうですか．ちょっとくどいかな？　“大きさ”は長さだけの意味に限定されないので，ちょっと気になるのです．

大多数の日本語フォント【では→において、】長さの相対単位として全幅は全角に一致する。

**Reply by kidayasuo** [reply]
### 文字の外枠
“文字の外枠”という用語の経緯、ありがとうございます。これからの時代の日本語版の読者層は、日本語組版のみならず、英語組版も理解すべき人たちですので、用語はできるだけ共通の方が受け入れられやすく、理解も進みやすいかと思います。必ずしもカタカナにすれば良いと考えているわけではないのですが。

### 全角
正方形の違和感については理解しました。確かに正方形の大きさというと、面積にも取れますね。

ただ、あくまで想像ではありますが、「全角」が純粋に長さの単位であって、それが正方形のボディを前提している、という順番ではなくて、元々「全角」は「角」という言葉の示すように、まず活字の占める正方形のことであって、その幅を以て長さの基準となした、という順番の方が、その名前（角）や用語の全般的な使われ方（例えば全角文字）から考えてしっくりきます。語源や用法の変遷についての研究があると良いと思うのですが、見つけられませんでした。

### フォントでは vs フォントにおいて
私はよりシンプルな「では」を好むのですが、「において」とする方が良いでしょうか？　確かに「において」は漢文的表現で格調高く、「で」はカジュアルですね。

### 相対単位
相対単位、を加えられた意味は、サイズに相対だよ、ということですよね。これ私も悩んでいます。この幅はサイズ依存であって、よって「その場に使用されているフォントに含まれる和字間隔（U+3000）の、**使用されているサイズにおける**、字詰め方向の幅」などと言うべきなのかもしれません。ちょっと長いというか、サイズにおけるは自明なので説明しなくても良いかなと思い省いたのですが。定義の部分だけはきちんとして、あとは省くのはどうでしょう？


アップデートしました。[そのバージョン](https://github.com/w3c/jlreq-d/blob/9e2b3a612f7950f08612283c5cb4b2295d01d347/drafts/.全角に替わる単位.md)。

**Comment by KobayashiToshi** [comment]
“では”は，“…では，…では”と連続するのが気になっただけです．

相対単位ということは，特にいわなくてもよいと思います．ある意味，私の口癖かもしれない．

> それらのフォントに対して全幅は全角に一致する。

でいいように思うのですが，
全角＝a：正方形のボディ（又は文字の外枠）＋b：文字が正方形（又は文字の外枠）のボディもっていること
全幅＝そこでの和字間隔の字幅

多くのフォントは正方形ですが，あくまで“全幅”は幅を示すということですからaだけが一致することです．いわずもなことといえば，その通りですが，一瞬，全幅はbも連想してしまう．その点で，一致するのはaだけよ，といえば問題がないように思うのですが，

**Reply by kidayasuo** [reply]
「全幅と全角の長さは一致する」にしてみました。いかがでしょう。

また、その段落の記述全体がどうも満足行かなかったので、書き直してみました。見ていただけると嬉しいです。
https://github.com/w3c/jlreq-d/blob/dbaf1612ee976d5faf40a062f1866b2ec8af36ba/drafts/.全角に替わる単位.md

**Comment by himorin** [comment]
closing per https://github.com/w3c/jlreq-d/pull/132
check current draft at https://github.com/w3c/jlreq-d/blob/gh-pages/drafts/%E5%85%A8%E5%B9%85%E3%81%AE%E5%AE%9A%E7%BE%A9.md
